File: frontend/src/types/execution.ts
export interface RunResponse {
  run_id:  string;
  status:  string;
  task_id: string;
}

export interface RunSummary {
  run_id:       string;
  workspace_id: string;
  /** "completed" | "running" | "error" */
  status:       string;
  created_at:   string;
}

export interface ExecutionDetail {
  node_id:       string;
  node_name:     string;
  /** "PENDING" | "RUNNING" | "SUCCESS" | "ERROR" | "SKIPPED" */
  status:        string;
  input_data:    Record<string, unknown> | null;
  output_data:   Record<string, unknown> | null;
  error_message: string | null;
  duration_ms:   number | null;
}

export interface RunDetail {
  run_id:       string;
  workspace_id: string;
  status:       string;
  executions:   ExecutionDetail[];
  created_at:   string;
}

File: backend/app/schemas/execution.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class RunRequest(BaseModel):
    initial_context: dict = {}
    node_id: uuid.UUID | None = None  # if set, run only this node


class RunResponse(BaseModel):
    run_id: uuid.UUID
    status: str  # always "queued"
    task_id: str


class ExecutionDetail(BaseModel):
    node_id: uuid.UUID
    node_name: str
    status: str
    input_data: dict | None
    output_data: dict | None
    error_message: str | None
    duration_ms: int | None


class RunSummary(BaseModel):
    run_id: uuid.UUID
    workspace_id: uuid.UUID
    status: str  # "running" | "completed" | "error"
    created_at: datetime


class RunDetail(BaseModel):
    run_id: uuid.UUID
    workspace_id: uuid.UUID
    status: str
    executions: list[ExecutionDetail]
    created_at: datetime
